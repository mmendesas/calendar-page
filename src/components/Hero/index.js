import React from 'react';
import PropTypes from 'prop-types';

import {
  Container,
  Image,
  Info,
  Flag,
  InfoContent,
  Title,
  TextContent,
  Button,
} from './styles';

export default function Hero({ data }) {
  const { image, title, datetime, deadline } = data;
  return (
    <Container data-testid="hero">
      <Image alt="Hero" url={image} />
      <Info>
        <Flag name="canada" />
        <InfoContent>
          <Title>{title}</Title>
          <div>
            <TextContent>
              <span>Vancouver - Canada</span>
              <span>{datetime}</span>
              <span>
                Deadline: <span>{deadline}</span>
              </span>
            </TextContent>
            <Button>See Application</Button>
          </div>
        </InfoContent>
      </Info>
    </Container>
  );
}

Hero.propTypes = {
  data: PropTypes.shape({
    image: PropTypes.string,
    title: PropTypes.string,
    datetime: PropTypes.string,
    location: PropTypes.shape({
      city: PropTypes.string,
      country: PropTypes.string,
    }),
    deadline: PropTypes.string,
  }).isRequired,
};
