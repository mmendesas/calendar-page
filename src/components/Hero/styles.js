import styled from 'styled-components';

export const Container = styled.div`
  max-width: 1280px;
  width: 100%;

  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 6px 0px,
    rgba(0, 0, 0, 0.1) 0px 2px 12px 0px !important;
  border-radius: 4px;
`;

export const Image = styled.div`
  background: url(https://vanhackblobstorageprod.blob.core.windows.net/img/events/cover/relocation-summit-2019.png)
    no-repeat center;
  background-size: cover;
  width: 100%;
  height: 320px;
`;

export const Info = styled.div`
  display: flex;
  flex-direction: row;
  padding: 15px;

  align-items: center;
`;

export const Flag = styled.div`
  background: url(https://vanhackblobstorageprod.blob.core.windows.net/flags/${props => props.name || 'brazil'}.svg)
    no-repeat center;
  background-size: cover;
  width: 50px;
  height: 50px;
  margin: 0px 20px;
  border-radius: 50%;
`;

export const InfoContent = styled.div`
  padding: 15px;
  width: 100%;

  div {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }
`;

export const Title = styled.h2`
  padding-bottom: 10px;
`;

export const TextContent = styled.div`
  display: flex;
  justify-content: center;
  font-size: 16px;
  span {
    &:nth-child(2) {
      font-weight: bold;
    }

    & + span {
      margin-left: 30px;
    }

    span {
      font-weight: bold;
    }
  }
`;

export const Button = styled.button`
  color: #3c74cf;
  border: 1px solid rgba(60, 116, 207, 0.5);
  padding: 8px 24px;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 4px;
  line-height: 1.75;
  background-color: transparent;

  &:hover {
    border-color: #3c74cf;
    background-color: rgba(60, 116, 207, 0.08);
  }
`;
