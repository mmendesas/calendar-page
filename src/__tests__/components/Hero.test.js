import React from 'react';
import { render } from '@testing-library/react';

import Hero from '~/components/Hero';

describe('[Component] Hero', () => {
  const setup = () => {
    const info = {
      title: 'Some Test',
      datetime: 'November 15 - 20, 2019',
      location: {
        city: 'Vancouver',
        country: 'Canada',
      },
      deadline: '11/20/2019',
    };
    const utils = render(<Hero data={info} />);

    return {
      info,
      utils,
    };
  };

  it('should be match snapshot', () => {
    const { utils } = setup();
    const { getByTestId } = utils;
    expect(getByTestId('hero')).toMatchSnapshot();
  });

  it('should render Hero with properties', () => {
    const { utils, info } = setup();
    const { getByText, debug } = utils;
    const { city, country } = info.location;

    Object.values(info).forEach(prop => {
      const text = typeof prop !== 'string' ? `${city} - ${country}` : prop;
      expect(getByText(text)).toBeTruthy();
    });
  });
});
