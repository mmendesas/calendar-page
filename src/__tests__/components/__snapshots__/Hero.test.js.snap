// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Hero should be match snapshot 1`] = `
<div
  class="sc-bdVaJa deTEtB"
  data-testid="hero"
>
  <div
    alt="Hero"
    class="sc-bwzfXH caOQhJ"
  />
  <div
    class="sc-htpNat hkTRzL"
  >
    <div
      class="sc-bxivhb bHkRbh"
      name="canada"
    />
    <div
      class="sc-ifAKCX jKojjw"
    >
      <h2
        class="sc-EHOje dDmdei"
      >
        Some Test
      </h2>
      <div>
        <div
          class="sc-bZQynM gBZrhZ"
        >
          <span>
            Vancouver - Canada
          </span>
          <span>
            November 15 - 20, 2019
          </span>
          <span>
            Deadline: 
            <span>
              11/20/2019
            </span>
          </span>
        </div>
        <button
          class="sc-gzVnrw lmmXHA"
        >
          See Application
        </button>
      </div>
    </div>
  </div>
</div>
`;
